apiserver/facades/client/client/status.go

```diff
@@ -546,32 +555,60 @@ type statusContext struct {
 // machine and machines[1..n] are any containers (including nested ones).
 //
 // If machineIds is non-nil, only machines whose IDs are in the set are returned.
-func fetchMachines(st Backend, machineIds set.Strings) (map[string][]*state.Machine, error) {
-	v := make(map[string][]*state.Machine)
+func (context *statusContext) fetchMachines(st Backend) error {
+	if context.model.Type() == state.ModelTypeCAAS {
+		return nil
+	}
+	context.machines = make(map[string][]*state.Machine)
+	context.allMachines = make(map[string]*state.Machine)
+
 	machines, err := st.AllMachines()
 	if err != nil {
-		return nil, err
+		return err
 	}
 	// AllMachines gives us machines sorted by id.
 	for _, m := range machines {
-		if machineIds != nil && !machineIds.Contains(m.Id()) {
-			continue
-		}
-		parentId, ok := m.ParentId()
+		context.allMachines[m.Id()] = m
+		_, ok := m.ParentId()
 		if !ok {
 			// Only top level host machines go directly into the machine map.
-			v[m.Id()] = []*state.Machine{m}
+			context.machines[m.Id()] = []*state.Machine{m}
 		} else {
 			topParentId := state.TopParentId(m.Id())
-			machines, ok := v[topParentId]
-			if !ok {
-				panic(fmt.Errorf("unexpected machine id %q", parentId))
-			}
-			machines = append(machines, m)
-			v[topParentId] = machines
+			machines := context.machines[topParentId]
+			context.machines[topParentId] = append(machines, m)
 		}
 	}
-	return v, nil
+
+	context.allInstances, err = context.model.AllInstanceData()
+	if err != nil {
+		return err
+	}
+	context.allConstraints, err = context.model.AllConstraints()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (context *statusContext) fetchOpenPorts(st Backend) error {
+	if context.model.Type() == state.ModelTypeCAAS {
+		return nil
+	}
+	context.openPorts = make(map[string]*state.Ports)
+	allOpenPorts, err := context.model.AllPorts()
```

Comment:
There are abstractions for ports/portranges in the network package. We should probably be using those for new code in facades instead of `state.Port`

Reply:
Agreed